### PR TITLE
Update `Roadmap`, add `About` pages

### PR DESCRIPTION
**Pull request by nonparibus**
## 🌟 Purpose

Ahead of rendering active (and past/accepted RFCs) on the Þ website, we want to clean up the current "RFCs and Roadmap" page somewhat.

## Changes

- [x] Updates the RFCs template (fixes/minor tweaks)
- [x] Fixes a broken link in the old `rfcs` directory README
- [x] Updates the roadmap page copy (just waiting on 'history' section)
- [x] Updates the roadmap page slug and redirects the old location to the new one
- [x] Updates the slugs for the module specifications and adds redirects from old to new locations
- [x] Introduces copy for a new `/about` page
- [x] Adds Next.js routes for showing the new `/about` page on the live Þ site (see https://github.com/blockprotocol/blockprotocol/pull/1077 for recent example)
- [x] Add support for `h6` heading styles, now required on the roadmap page

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204193856002814
  - https://app.asana.com/0/0/1204160628104486

### CODE REVIEW

File: apps/site/src/theme/components/data-display/mui-typography-theme-options.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const MuiTypographyThemeOptions: Components<Theme>["MuiTypography"] = {
       bpHeading3: "h3",
       bpHeading4: "h4",
       bpHeading5: "h5",
+      bpHeading5: "h6",
```
Comment (CiaranMn) [inline apps/site/src/theme/components/data-display/mui-typography-theme-options.ts:13]:
```suggestion
      bpHeading6: "h6",
```

File: apps/site/src/_pages/docs/6_roadmap.mdx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+- **Version 0.2 and Graph Module:** in June 2022, version 0.2 of the specification [launched at WeAreDevelopers](https://hash.dev/blog/block-protocol-v02) in Berlin. This introduced the modular architecture used by the protocol today, including an initial release of the _graph module_, improving support for structured data in blocks separated from its logic.
+- **Hook Module:** in October 2022, the _hook module_ was added to the specification, allowing blocks to [integrate tightly with the native capabilities](https://hash.dev/blog/hook-service) of embedding applications, in an application non-specific way. For example, this allowed blocks with text fields to share the shortcuts, commands and text-styling/controls alreay present within their environment.
+- **Version 0.3:** in February 2023, [we released](https://hash.dev/blog/block-protocol-0-3-wordpress) the biggest update to the Block Protocol yet, overhauling the _core_ and _graph module_, and introducing the new _type system_.
```
Comment (CiaranMn) [inline apps/site/src/_pages/docs/6_roadmap.mdx:20]: This update didn't materially affect `core` 
```suggestion
- **Version 0.3:** in February 2023, [we released](https://hash.dev/blog/block-protocol-0-3-wordpress) the biggest update to the Block Protocol yet, overhauling the _graph module_, and introducing the new _type system_.
```

File: apps/site/src/theme/components/data-display/mui-typography-theme-options.ts (hunk 1 above)
Comment (nonparibus) [reply apps/site/src/theme/components/data-display/mui-typography-theme-options.ts]: Addressed in https://github.com/blockprotocol/blockprotocol/pull/1183/commits/47536be3c743ca001960adf17321d1ddebdaed16

**Review by CiaranMn — APPROVED** [review]

**Comment by erlend-sh** [comment]
> Learn more on our [RFCs & Roadmap](https://blockprotocol.org/docs/spec/rfcs_and_roadmap?utm_medium=organic&utm_source=github_readme_blockprotocol-repo_root) page in the docs.

This link doesn't work.

**Pull request merged**